### DISCUSSION TITLE
TM1 logins and logouts

### DISCUSSION THREAD

**Discussion by raulpkd** [cubewise-code/tm1py#983] [Q&A] [answered]
Hi, I'm having a problem that while running an analytics program with a large dataset. The problem is that every request creates a login and login note to the login log. You can imagine the massive amount of rows being created. I wonder if it is possible to some use the same login session for multiple sessions, create a batch file for the requests or something else? This could reduce the stress on the login logging.

**Comment by MariusWirtz** [comment, marked as answer]
When you have multiple interactions with TM1 _(e.g. data retrieval or writeback)_ from the same Python script there is no need to login for each operation separately.
Here is an example.

``` python
from TM1py.Services.TM1Service import TM1Service

tm1_params = {...}

with TM1Service(**tm1_params) as tm1:
    cells = {('e1', 'e2'): 1}
    tm1.cells.write_values("c1", cells)

    cells = {('e1', 'e2'): 2}
    tm1.cells.write_values("c1", cells)
    
    cells = {('e1', 'e2'): 3}
    tm1.cells.write_values("c1", cells)
```

Does that answer your question?
